### PR TITLE
fix: persist option none is not working

### PR DESCRIPTION
**Pull request by snax4a**
I have fixed persist options.
Before that, the `'none'` option was not working. Now, when `'none'` option is passed the state is not being saved to any store in action handler.

I have introduced PersistOption enum for better DX but string literals are also available for backward compatibility.
```ts
export enum PersistOption {
  None = 'none',
  OnAction = 'onAction',
  BeforeUnload = 'beforeUnload',
}
```

CleanShot 2022-09-16 at 14 45 23

I have also updated the store factory options type so it's less hacky.

Locally everything seems to work correctly

### CODE REVIEW

File: src/StateMachineContext.tsx [hunk 1]
```diff
@@ -15,9 +15,9 @@ export const StateMachineProvider: React.FC<PropsChildren> = ({ children }) => {
   const [state, setState] = React.useState(storeFactory.state);
 
   React.useEffect(() => {
-    if (storeFactory.options.persist === PERSIST_BEFORE_UNLOAD) {
+    if (storeFactory.options.persist === PersistOption.BeforeUnload) {
       window.onbeforeunload = () => storeFactory.saveStore();
-      storeFactory.options.storageType.removeItem(storeFactory.options.name);
+      storeFactory.options.storageType?.removeItem(storeFactory.options.name!);
```
Comment (bluebill1049) [inline src/StateMachineContext.tsx:20]: could we avoid the optional chain operator? it increases the package size during transpile.

File: src/constants.ts [hunk 2]
```diff
@@ -1,3 +1,8 @@
 export const STORE_DEFAULT_NAME = '__LSM__';
 export const STORE_ACTION_NAME = '__LSM_NAME__';
-export const PERSIST_BEFORE_UNLOAD = 'beforeUnload';
+
+export enum PersistOption {
+  None = 'none',
+  OnAction = 'onAction',
+  BeforeUnload = 'beforeUnload',
+}
```
Comment (bluebill1049) [inline src/constants.ts:8]:
```suggestion
export const PersistOption = {
  none: 'none',
  action: 'action',
  unload = 'unload',
} as const
```

save some bytes over enum

File: src/index.ts [hunk 3]
```diff
@@ -1,5 +1,12 @@
 import { StateMachineProvider } from './StateMachineContext';
 import { createStore, useStateMachine } from './stateMachine';
 import { GlobalState } from './types';
+import { PersistOption } from './constants';
```
Comment (bluebill1049) [inline src/index.ts:4]: maybe we can skip exporting this?

**Review by bluebill1049 — COMMENTED** [review]
thanks very much for the PR, some minor comments.

File: src/index.ts (hunk 3 above)
Comment (snax4a) [reply src/index.ts:4]: Sure, I will slightly tweak it to meet all your requirements and will push the changes soon

File: src/constants.ts (hunk 2 above)
Comment (snax4a) [reply src/constants.ts:8]: So we don't need backward compatibility here and we can use these shorter option names? "none", "action", "unload"?

**Review by bluebill1049 — APPROVED** [review]
Awesome new feature! thanks for the contribution.

**Pull request merged**